File: Two Sum/Two Sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        cache = dict()

        for i, n in enumerate(nums):
            if target - n in cache:
                return [cache[target-n], i]

            cache[n] = i

        return [-1,-1]

File: Two Sum/Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> numToIndex;

        for(int i = 0; i < nums.size(); i++)
        {
            int needed = target - nums[i];
            
            if(numToIndex.count(needed) > 0)
            {
                return {numToIndex[needed], i};
            }
            numToIndex[nums[i]] = i;
        }

        return {};
    }
};